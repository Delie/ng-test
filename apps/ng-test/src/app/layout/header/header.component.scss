.main-header {
  background: #238ccc;
  color: #fff;
  padding: 10px;
  font-weight: bold;
  font-size: 40px;
  margin-bottom: 1px;
}

.sub-header {
  background: #555;
  padding: 10px;
  color: #fff;
}
