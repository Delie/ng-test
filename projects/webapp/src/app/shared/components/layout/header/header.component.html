<div class="main-header">
  @ng-test/webapp
</div>
<div class="sub-header">
  <a [routerLink]="['/']">{{ 'Home' | translate }}</a> |
  <a [routerLink]="['/comments']">{{ 'Comments' | translate }}</a>
</div>
