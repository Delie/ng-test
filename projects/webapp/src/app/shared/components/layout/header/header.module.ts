import { NgModule } from '@angular/core';
import { HeaderComponent } from './header.component';
import { RouterModule } from '@angular/router';
import { TranslateModule } from '@ngx-translate/core';

@NgModule({
  declarations: [HeaderComponent],
  imports: [RouterModule, TranslateModule],
  exports: [HeaderComponent]
})
export class HeaderModule {}
