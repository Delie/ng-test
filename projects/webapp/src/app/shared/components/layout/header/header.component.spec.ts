import { async } from '@angular/core/testing';
import { Shallow } from 'shallow-render';
import { HeaderComponent } from './header.component';
import { HeaderModule } from './header.module';

describe('HeaderComponent', () => {
  let shallow: Shallow<HeaderComponent>;
  beforeEach(async(() => {
    shallow = new Shallow(HeaderComponent, HeaderModule);
  }));

  it('should create the component', async () => {
    const { element } = await shallow.render();
    expect(element.nativeElement).toMatchSnapshot();
  });
});
