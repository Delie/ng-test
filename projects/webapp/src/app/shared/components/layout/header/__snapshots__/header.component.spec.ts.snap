// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should create the component 1`] = `
<app-header>
  <div
    class="main-header"
  >
     @ng-test/webapp

  </div>
  <div
    class="sub-header"
  >
    <a
      ng-reflect-router-link="/"
    >
      
    </a>
     | 
    <a
      ng-reflect-router-link="/comments"
    >
      
    </a>
  </div>
</app-header>
`;
